### DISCUSSION TITLE
Problem with getFullList

### DISCUSSION THREAD

**Discussion by VictorDevCode21** [pocketbase/pocketbase#5371] [Q&A] [answered]
I am trying to get the list of all the orders on my orders collection, but returns an empty array, I know it's because of the API rules, I have this rule for list/search: @request.auth.id = user.id, but when I'm using the getFullList, doesn't let me pass the id trough params. This is the code:  

const result = await pb.collection('orders').getFullList<Order>(
                    {
                    sort: '-created',
                }

.

**Comment by ganigeorgiev** [comment]
It is not clear what do you mean by _"doesn't let me pass the id trough params"_.

The rule `@request.auth.id = user.id` (_or you can use directly `@request.auth.id = user`_) means that only the records where the authorized user id matches with the id of the `user` relation field will be returned. If you have a client-side filter, then it is applied in addition to the one from the API rules.

Or in other words, you don't have to pass anything and you simply have to be authenticated to retrieve the records associated with your user:

```js
await pb.collection("users").authWithPassword("test@example.com", "123456")

console.log(pb.authStore.token) // verify that it is not empty

const result = await pb.collection("orders").getFullList({ sort: "-created" })
```

**Reply by VictorDevCode21** [reply]
And how can I return every record? I am doing a panel and I need that just by verifying that the user exists, the panel shows every order. Not just the orders created by that user in particular.

**Reply by ganigeorgiev** [reply, marked as answer]
I think there is a language barrier that prevents me to understand what exactly you are trying to do.

If you want to allow access to **all authenticated users** to list any record from the "orders" collection, then you can change your "orders" collection List API rule to `@request.auth.id != ""`.

Then to filter the record by any user id you can do something like:

```js
await pb.collection("users").authWithPassword("test@example.com", "123456")

const result = await pb.collection("orders").getFullList({
    // assuming that "user" is the name of the relation field in your orders collection
    // you can replace it with the name of any other field from your collection
    filter: `user = "anyUserId"`,
    sort: "-created",
})
```

I find myself replying with this frequently, but please don't treat the discussions as llm chat and try to elaborate a little more on your use case and collections structure to understand the bigger picture because neither me, nor anyone else, has context what you are doing.

**Reply by VictorDevCode21** [reply]
Thanks, I think that was the answer that I needed.
